iceoryx_hoofs/container/include/iox/detail/vector.inl

```diff
@@ -52,10 +53,13 @@ inline vector<T, Capacity>::vector(const uint64_t count) noexcept
     }
 
     m_size = std::min(count, Capacity);
-    for (uint64_t i{0U}; i < m_size; ++i)
+    if constexpr (!std::is_trivial<T>::value)
     {
-        // AXIVION Next Line AutosarC++19_03-A18.5.2, FaultDetection-IndirectAssignmentOverflow : False positive, it is a placement new. Size guaranteed by T.
-        new (&at(i)) T();
+        for (uint64_t i{0U}; i < m_size; ++i)
+        {
+            // AXIVION Next Line AutosarC++19_03-A18.5.2, FaultDetection-IndirectAssignmentOverflow : False positive, it is a placement new. Size guaranteed by T.
+            new (&at(i)) T();
+        }
```

Comment:
Oh, it just occurred to me that `resize` might also need to have a variant with `UnsafeUninitialized`

Reply:
@elBoberido Safety over performance. To achieve this, I think your proposal with the three constructors seems reasonable. 
I'm not aware of an `UnsafeUninitialized` kind of type in our codebase.

Reply:
Here, following your guidelines, I see 3 solutions:
 - use ```is_class``` to avoid initialized data on simple data
 - keep all as is, except for the ```operator=```, using ```memcpy``` when the data is ```is_trivially_copyable```
 - pass to the constructor and resize an additional param to skip initialization only if the data ```is_trivial```

What do you like?

Reply:
I Got it!
If you agree I could update this PR with the first solution and implement SFINAE or something similar in a second PR.

Reply:
Yes, go ahead. SFINEA should be a oneliner. We can decide once the PR is updated. If it becomes more difficult than a oneliner we can postpone it to a follow up PR.

Reply:
Ok!